src/app/common/hooks/use-submit-stx-transaction.ts

```diff
@@ -45,14 +44,12 @@ export function useSubmitTransactionCallback({ loadingKey }: UseSubmitTransactio
             onClose();
             setIsIdle();
           } else {
-            const txid = `0x${response.txid}`;
-            if (!externalTxid.includes(txid)) {
-              await setLocalTxs({
+            dispatch(
+              submittedTransactionsActions.transactionAdded({
                 rawTx: transaction.serialize().toString('hex'),
-                timestamp: todaysIsoDate(),
-                txid,
-              });
-            }
+                txId: safelyFormatHexTxid(response.txid),
+              })
```

Comment:
I'd create a hook that exposes a method calling this dispatch (and other actions from this slice)